### PR TITLE
Fix clip histogram infinite loop

### LINKED ISSUE #1761: Infinite loop in exposure.equalize_adapthist()
The `equalize_adapthist()` function can hang, caused by the following:

When using a very small clip_limit parameter (e.g. 0.001) on a grayscale image with low contrast, the function `clip_histogram()` in `_adapthist.py` hangs in an infinite loop because the `n_excess` variable is not being lowered in the `while n_excess > 0` loop.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #1761.

### CODE REVIEW

File: skimage/exposure/_adapthist.py [hunk 1]
```diff
@@ -244,7 +244,8 @@ def clip_histogram(hist, clip_limit):
     n_excess -= mid.size * clip_limit - mid.sum()
     hist[mid_mask] = clip_limit
 
-    while n_excess > 0:  # Redistribute remaining excess
+    iteration = 0
+    while n_excess > 0 and iteration < 20:  # Redistribute remaining excess
```
Comment (jni) [inline skimage/exposure/_adapthist.py]: How did you choose 20? I'm uncomfortable with it being hardcoded like this. At the very least, I would create a variable `max_iterations` and with it a comment about why it exists and how its value was chosen.

Comment (JDWarner) [inline skimage/exposure/_adapthist.py]: Agreed. Also, as I mentioned in the Issue previously, I think the problem may resolve without modification of the underlying code if we simply check and ensure `clip_level` is greater than the image minimum value.

**Comment by blink1073** [comment]
Fixed, thanks for the reality check.

**Comment by jni** [comment]
Brilliant! This is way better. The system works, baby! =D

**Pull request merged**